### PR TITLE
[fix] #4014: Use method(not field) access in client

### LINKED ISSUE #4014: Fix client build
Client cannot be built with `cargo build -p iroha_client` again due to recent changes in #3929

### PR DESCRIPTION
**Pull request by mversic**
### Linked issue

Closes #4014

### Checklist

- [ ] I've read `CONTRIBUTING.md`
- [ ] I've used the standard signed-off commit format (or will squash just before merging)
- [ ] All applicable CI checks pass (or I promised to make them pass later)
- [ ] (optional) I've written unit tests for the code changes
- [ ] I replied to all comments after code review, marking all implemented changes with thumbs up

### CODE REVIEW

File: data_model/src/query/mod.rs [hunk 1]
```diff
@@ -176,7 +176,7 @@ pub mod model {
 
     /// Query with parameters client can specify.
     #[derive(
-        Debug, Constructor, Getters, Clone, PartialEq, Eq, Encode, Decode, Serialize, Deserialize,
+        Clone, Debug, PartialEq, Eq, Constructor, Getters, Encode, Decode, Serialize, Deserialize,
```
Comment (Arjentix) [inline data_model/src/query/mod.rs:179]: So is our policy to place std derives first? Didn't know about that

**Review by Arjentix — APPROVED** [review]

File: data_model/src/query/mod.rs (hunk 1 above)
Comment (mversic) [reply data_model/src/query/mod.rs:179]: there is no strict policy. If you look around you will see this pattern

File: client/src/client.rs [hunk 2]
```diff
@@ -387,9 +389,9 @@ impl QueryRequest {
 
         match self.request {
             iroha_data_model::query::QueryRequest::Query(query_with_params) => builder
-                .params(Vec::from(query_with_params.sorting))
-                .params(Vec::from(query_with_params.pagination))
-                .body(query_with_params.query),
+                .params(Vec::from(query_with_params.sorting().clone()))
```
Comment (DCNick3) [inline client/src/client.rs:392]: Not introduced in this PR, but the conversion of `Pagination` and `Sorting` structs to `Vec` is a bit unintuitive.. I feel like explicitly naming it something like `into_request_params`.

Maybe even make it into a `as_request_params` & make it accept a reference (it's cloned here anyways).

Comment (mversic) [reply client/src/client.rs:392]: opened #4017 4017

**Review by DCNick3 — APPROVED** [review]

**Pull request merged**